skiplang/compiler/src/sklib.sk

```diff
@@ -71,75 +95,32 @@ class Metadata(
   }
 }
 
-// The Sklib class does not contain the possible object files present in the
-// archive as those are handled directly by the linker.
-mutable class .Sklib(
-  ar: mutable ARParser.Archive<mutable IO.File>,
-  sources: Map<String, ARParser.Header>,
-  metadata: Metadata,
-) {
-  const kMetadataFile: String = "lib.skmeta";
-
-  static fun read(
-    file: mutable IO.File,
-  ): Result<mutable Sklib, ARParser.Error> {
-    ar = ARParser.Archive::read(file) match {
-    | Success(ar) -> ar
-    | Failure(err) -> return Failure(err)
-    };
-
-    metadata = Metadata::read(
-      (ar[static::kMetadataFile] match {
-      | Success(bytes) -> bytes
-      | Failure(err) -> return Failure(err)
-      } match {
-      | Some(bytes) -> bytes
-      | None() -> "".bytes()
-      // TODO: Properly fail.
-      }),
-    );
-
-    sources = mutable Map[];
-    for (header in ar) {
-      if (!header.identifier.endsWith(".sk")) continue;
-      sources.set(header.identifier.replace("#", "/"), header)
-    };
-
-    Success(mutable Sklib(ar, sources.chill(), metadata))
-  }
-
-  mutable fun read_file(path: String): Result<String, ARParser.Error> {
-    this.ar[path.replace("/", "#")] match {
-    | Success(Some(bytes)) -> Success(String::fromUtf8(bytes.collect(Array)))
-    | Success(None()) -> Success("")
-    | Failure(err) -> Failure(err)
-    }
-  }
-
-  static fun read_metadata(path: String): Result<Metadata, ARParser.Error> {
-    file = IO.File::open(path, IO.OpenOptions{read => true});
-    ar = ARParser.Archive::read(file) match {
-    | Success(ar) -> ar
+const kMetadataFile: String = "lib.skmeta";
+
+fun read_metadata(path: String): Result<Metadata, ARParser.Error> {
+  file = IO.File::open(path, IO.OpenOptions{read => true});
+  ar = ARParser.Archive::read(file) match {
+  | Success(ar) -> ar
+  | Failure(err) -> return Failure(err)
+  };
+  res = Metadata::read(
+    (ar[kMetadataFile] match {
+    | Success(bytes) -> bytes
     | Failure(err) -> return Failure(err)
-    };
-    res = Metadata::read(
-      (ar[static::kMetadataFile] match {
-      | Success(bytes) -> bytes
-      | Failure(err) -> return Failure(err)
-      } match {
-      | Some(bytes) -> bytes
-      | None() -> "".bytes()
-      // TODO: Properly fail.
-      }),
-    );
-    file.close();
-    Success(res)
-  }
+    } match {
+    | Some(bytes) -> bytes
+    | None() -> "".bytes()
+    // TODO: Properly fail.
```

Comment:
Did you mean to do this before landing this PR, or maybe change this to FIXME?

Reply:
This was copied over from the previous version of the code, but agreed that it should be a `FIXME`.